### DISCUSSION TITLE
Can add mono version branch to "Properties→Betas" for Godot on Steam?

### DISCUSSION THREAD

**Discussion by Sonic853** [godotengine/godot-proposals#7551] [Editor]
### Describe the project you are working on

I am using C# for game development.

### Describe the problem or limitation you are having in your project

There is no mono version of Godot on Steam.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add mono version branch to "Properties→Betas" for Godot on Steam
![Snipaste_2022-05-25_19-55-35](https://user-images.githubusercontent.com/8389962/170256769-aee9f729-5753-4761-bd22-12a2d60c65f6.png)

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

When developers select the mono branch in "Properties→Betas", they can get the latest mono version of Godot.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Developers can choose to download the Godot mono version from the Godot website or Scoop, but it is very inconvenient to update Godot.

### Is there a reason why this should be core and not an add-on in the asset library?

This will make it easier for developers to upgrade and use Godot mono version.

**Comment by FrostBird347** [comment]
It would also be nice to add a C# download option on itch.io for the same reasons

**Reply by roke-julian-lockhart** [reply]
> https://github.com/godotengine/godot-proposals/discussions/7551#discussioncomment-6814635

@FrostBird347, I believe that that should be filed as a separate request.

**Reply by FrostBird347** [reply]
Considering what was said at https://github.com/godotengine/godot-proposals/discussions/7551#discussioncomment-8231226, I see little reason to create a new request.

**Comment by roke-julian-lockhart** [comment]
> https://github.com/godotengine/godot-proposals/issues/4589#event-10183000643

@YuriSizov, this shouldn't be solely a discussion. It prevents users subscribing solely to issue closure. Leave the issue open, but allow this place to be used for discussion of the feature, please.

**Reply by AThousandShips** [reply]
There wasn't any specific details, that's required for a proposal, and it can't be reopened, please open a new proposal *if you have a description for how to accomplish this*, with more specifics 🙂

In either case this was converted over a and a half ago so not really relevant now, and no one has provided details or ideas so far

**Reply by roke-julian-lockhart** [reply]
> https://github.com/godotengine/godot-proposals/discussions/7551#discussioncomment-8224528

@AThousandShips, it's relevant to me, because I can't easily use the C# version except via Steam. Regardless, what exact specifics are mandatory? What you've stated is too vague for it to be actionable on my behalf.

**Reply by AThousandShips** [reply]
No, you're not listening to me... The old issue report isn't relevant, and I'm being very specific, follow the proposal format, if you don't think a discussion thread is enough, open a proposal instead, that's all

The old post *cannot* be reopened, it is impossible, not that it *won't*, it's not possible to do so because of how GitHub works

But there's really no reason this needs to be an issue instead of a discussion, it's not gotten any comments since it's been a discussion and no reason to believe it had gotten more otherwise

(Also when responding directly to people there's no need to tag them, it's just annoying, so please don't when you're just in a conversation)

**Reply by YuriSizov** [reply]
Well, that's a bummer that you cannot subscribe to a discussion being resolved like you can with issues. But we don't use issues for general ideas which are not directly actionable.

That said, we don't have plans to add C# builds as an optional download on Steam. We have plans to have a unified Godot editor so you don't need a separate download. This will resolve the issue for Steam users.

**Reply by roke-julian-lockhart** [reply]
> https://github.com/godotengine/godot-proposals/discussions/7551#discussioncomment-8231226

Indeed, @YuriSizov, I saw that on the 4.0 blog post (although not in 4.1). Does an issue tracking that work exist (yet)? If not, can I create one, or might I ask of you to? That really matters to me – I can't easily use Godot until then for myriad reasons.

**Reply by YuriSizov** [reply]
There isn't one, AFAIK, and there can't be one yet because this change requires several pre-requisite tasks to be completed first. So there is nothing to put into the issue as far as implementation details go.

Why do you need Steam to use Godot?

**Reply by roke-julian-lockhart** [reply]
> https://github.com/godotengine/godot-proposals/discussions/7551#discussioncomment-8232661

@YuriSizov, I suppose this is the kind of time we wish GitHub had proper issue correlation like Bugzilla does.

My preference for Steam is due to it providing a consistent installation base across operating systems, not least because DotNet isn't always trivial to install on every Linux distribution. I'd use a native RPM package for the mono version if one existed for openSUSE, but it doesn't. Dependencies change quick enough on rolling releases that using an unpackaged version isn't convenient.

**Comment by AyOhEe** [comment]
Very interested in this. I exclusively use the .net build of Godot, and use steam to track my hours because I like seeing the statistic and it lets my friends know when I'm working on something/when I'm free. I do not, however, like having to download Godot twice (once from the website for .net, once through steam for it to not replace it later) and then rename the .exe when an update to the engine releases.

I'd very much appreciate if this could be a thing, if it's not too much bother. Even a stable-latest-mono would be great!

**Comment by ElectroluxV2** [comment]
Hi,
If anyone wants a mono version on Steam **solely for time tracking purposes**, I found a pretty simple way of running a local mono version within Steam (installed manually, outside of steam).


1. Install Godot via Steam (version is not relevant as we will not use it anyway - yes this will waste ~2GB on your drive)
2. Go to properties (right click on Godot Engine within Library view)
3. In general tab specify inline script followed by `%command%`
    - Linux & macOS: `/bin/bash -c '/opt/homebrew/bin/godot-mono && while pgrep "Godot" ; do sleep 1.0 ; done' %command%`
    - Windows (not tested, sorry): `cmd /c "@echo off & start C:\some-path\godot_mono.exe -- %command% & :loop & tasklist | findstr /I \"godot_mono.exe\" > nul & if %errorlevel% equ 0 ( timeout /t 1 /nobreak > nul & goto loop )"`
4. Done - now Steam will launch the mono version and track time spent within the engine image

Reference screenshot: 
<img width="1028" alt="image" src="https://github.com/user-attachments/assets/26fb839d-dcfe-44d2-ad71-fb9633ead0a5" />

**Reply by ElectroluxV2** [reply]
@AyOhEe you may be interested in this

**Reply by AyOhEe** [reply]
Ah, that's cool. I've just been replacing the executable.

**Reply by unpWn4bl3** [reply]
windows version could be  `cmd /c "@echo off & start C:\some-path\godot_mono.exe %command% & :loop & tasklist | findstr /I \"godot_mono.exe\" > nul & if %errorlevel% equ 0 ( timeout /t 1 /nobreak > nul & goto loop )"`, which removes `--`.
